server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -188,14 +188,6 @@ public DesiredBalance compute(
             }
         }
 
-        // TODO must also bypass ResizeAllocationDecider
-        // TODO must also bypass RestoreInProgressAllocationDecider
-        // TODO what about delayed allocation?
-
-        // TODO consider also whether to unassign any shards that cannot remain on their current nodes so that the desired balance reflects
-        // the actual desired state of the cluster. But this would mean that a REPLACE shutdown needs special handling at reconciliation
-        // time too. But maybe it needs special handling anyway since reconciliation also tries to respect allocation rules.
```

Comment:
This should be handled in both compute and reconcile balance due to `org.elasticsearch.cluster.routing.allocation.decider.AllocationDecider#canRemain` calls